### PR TITLE
ogen: first draft of DSL API to alter or create the spec by code.

### PR DESCRIPTION
**Pull request by masseelch**
@ernado @a8m @rotemtam

You can see how to use the API in the test. I think this is a solid foundation. Opinions?

### CODE REVIEW

File: dsl_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+				SetEmail(ex.Info.Contact.Email).
+				Contact(),
+			).
+			SetLicense(LicenseBuilder().
+				SetName(ex.Info.License.Name).
+				SetURL(ex.Info.License.URL).
+				License()).
+			Info(),
+		).
+		Spec()
```
Comment (a8m) [inline dsl_test.go]: Do we need the `Builder` word there? Did you consider adding the setters to the actual types?

```go
s := spec.New().
    SetOpenAPI(ex.OpenAPI).
    SetInfo(
        spec.NewInfo().
            SetTitle(...).
            SetVersion(...).
            SetDescription(...),
    ).
    SetLicense(
        spec.NewLicense().
            SetURL(...).
            SetName(...),
    )
```

Maybe, drop the `New` word as well.

Comment (masseelch) [reply dsl_test.go]: Yeah, I did thought about that one and then changed it to have the builders decoupled from the spec itself. I see that this doesn't really make sense, true dat.

One reason as though, that the spec structs sometimes are pointers and sometimes not.

**Comment by ernado** [comment]
The most interesting part is how to define OpenAPI operations and schema for them :thinking:

**Comment by masseelch** [comment]
> The most interesting part is how to define OpenAPI operations and schema for them thinking

Jup. I currently sitting in front of it.  Will be part of the next PR. This was just to set the general API.

**Pull request merged**